libs/git-client/gradle/integrationTest.gradle.kts

```diff
@@ -0,0 +1,43 @@
+/*
+    Copyright 2017-2021 Charles Korn.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+val sourceSets = the<SourceSetContainer>()
+val mainAndTestCommon = sourceSets["main"].output + sourceSets["testCommon"].output
+
+val integrationTest by sourceSets.creating {
+    compileClasspath += mainAndTestCommon
+    runtimeClasspath += mainAndTestCommon
+}
+
+configurations["integrationTestImplementation"].extendsFrom(configurations["testImplementation"])
+configurations["integrationTestRuntimeOnly"].extendsFrom(configurations["testRuntimeOnly"])
+
+tasks.register<Test>("integrationTest") {
+    description = "Runs the integration tests."
+    group = "Verification"
+    testClassesDirs = integrationTest.output.classesDirs
+    classpath = integrationTest.runtimeClasspath
+
+    mustRunAfter("test")
+}
+
+tasks.named("check").configure {
```

Comment:
```suggestion
tasks.named("check") {
```

Reply:
AFAIK using `.configure` is the recommended practice to avoid eager realization of tasks, also see https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_pitfalls.

Reply:
The [documentation for `named()`](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-org.gradle.api.Action-) suggests that this variant is equivalent, and still does not trigger eager realisation of tasks.

To confirm this behaviour, I also tried a quick experiment with the following configuration:

```kotlin
tasks.register("task-1")
tasks.register("task-2")
tasks.register("task-3")
tasks.register("task-4")

tasks.named("task-1") {
    description = "First task description"
}

tasks.named("task-2").configure {
    description = "Second task description"
}

tasks.getByName("task-3") {
    description = "Third task description"
}
```

Running a build scan while running `task-4` shows that `task-3` is eagerly created during configuration, but `task-1` and `task-2` are not.

Reply:
Interesting, thanks for checking. I wonder if the Gradle docs about "task configuration avoidance pitfalls" should be adjusted then... anyway, I've removed the `.configure` calls now.